### DISCUSSION TITLE
how to add two esp32 board to one iPhone homekit?

### DISCUSSION THREAD

**Discussion by hzkincony** [HomeSpan/HomeSpan#475] [General]
i have download homespan to my first ESP32 board, it work well by iPhone homekit.
But if i want to add another esp32 board to homespan, it can't work. just two same ESP32 hardware board, how to do if i have 2,3,4,5 ESP32 boards? Thanks.

**Comment by HomeSpan** [comment]
You can use as many board as desired.  See the FAQ for details.

**Comment by hzkincony** [comment]
thanks for quick feedback. i have checked, can you send me link which is for my questions. my means: i have many same ESP32 board, not different products. all have same homespan firmware PCB board. such as if you have 5pcs of PCB, what need to do , so that all can work in homekit. Thanks.

**Comment by HomeSpan** [comment]
If you want to pair more than one board to the same HomeKit network (for example, if you are at home and use your iPhone to pair more than one ESP32), you need to make sure that the sketch on each ESP32 is programmed to have a unique Device ID, and unique HostName, and a unique Display Name.  Normally the Device ID is generated at random so will always be unique.  The Host Name is normally generated from the Device ID so will also be unique.  But the Display Name is something YOU set in your sketch, and needs to be different for every ESP32 board.  This is fully described on the HomeSpan Frequently Asked Questions [FAQ](https://github.com/HomeSpan/HomeSpan/blob/master/docs/FAQ.md) page.
